src/main/java/com/connexta/poller/service/Poller.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.poller.service;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class Poller {
+
+  private final StatusService statusService;
+  private final BlockingQueue<URI> transferQueue;
+  private ExecutorService executor;
+  private AtomicBoolean isStarted;
+
+  public Poller(StatusService statusService, BlockingQueue<URI> transferQueue) {
+    this.statusService = statusService;
+    this.transferQueue = transferQueue;
+    isStarted = new AtomicBoolean(false);
+  }
+
+  public void start() {
```

Comment:
I don't think we need these start and stop methods or need to keep track of whether it has been started. I think the Poller should be started when the application is started. Maybe you can implement a finalize method to define how we shutdown the executor.

Reply:
@connexta/i2o-cdr  If we let the objects call the StatusService directly, we won't even need the queue. That is the approach I'd like to pursue if the team is agreeable to that.